esp/esdllib/esdl_def_helper.cpp

```diff
@@ -249,9 +249,15 @@ void EsdlDefinitionHelper::toXSD( IEsdlDefObjectIterator &objs, StringBuffer &xs
     if( trans )
     {
         IProperties* params = *( parameters.getValue(xslId) );
-        const char *tns = (ns) ? ns :params->queryProp("tnsParam");
-
-        xml.appendf("<esxdl name=\"custom\" EsdlXslTypeId=\"%d\" xmlns:tns=\"%s\" ns_uri=\"%s\">", xslId, tns ? tns : "urn:unknown", tns ? tns : "urn:unknown");
+        StringBuffer tns((ns) ? ns :params->queryProp("tnsParam"));
+        // Since the xsl string params are quoted with ' we must remove
+        // it here before including it as the value of an attribute in
+        // the esxdl element
+        tns.stripChar('\'');
+        if( tns.isEmpty() )
+            tns.set("urn:unknown");
+
+        xml.appendf("<esxdl name=\"custom\" EsdlXslTypeId=\"%d\" xmlns:tns=\"%s\" ns_uri=\"%s\">", xslId, tns.str(), tns.str());
         this->toXML( objs, xml, version, opts, flags );
         xml.append("</esxdl>");
```

Comment:
This block was pretty simple and might not have warranted a stand-alone function, but the increased complexity may now justify a bit of refactoring to avoid deviations between the two occurrences.

Reply:
Created a class method to encapsulate the logic of determining which namespace to use and unquoting it if required.